Vue

ksyeo1010 opened the pull request:
- add vue interface

the difference here vs react/angular is that callback cant be changed/updated without reinit

laves commented on binding/vue/src/porcupine.ts:
```diff
@@ -12,37 +12,34 @@
+    options?: PorcupineOptions,
+  ) => Promise<void>,
+  start: () => Promise<void>,
+  pause: () => Promise<void>,
```
Just need to remove `pause` and add `release` here to match the interface we settled on

ksyeo1010 commented:
- Implemented interface. Had to do callbacks since variables are not updated. 
- updated demo to Vue3 (still works in vue 2, rhino/picovoice can still be Vue 2).

ErisMik commented on binding/vue/README.md:
````diff
@@ -1,127 +1,250 @@
+...
+```
+
+Add to the `options` object an `processErrorCallback` function if you would like
+to catch errors:
+
+```typescript
+methods: {
+  processErrorCallback: function (error: string) {
+  ...
+  }
+}
+
````
Is this necessary to mention since we already include a way to get the errors through the sdk?

ErisMik commented on binding/vue/README.md:
```diff
@@ -1,127 +1,250 @@
+
+### Release
+
+After done with Porcupine, either call `release` or destroy the mounted component:
```
```suggestion
When done with Porcupine, either call `release` or destroy the mounted component:
```

ErisMik commented on demo/vue/src/components/VoiceWidget.vue:
```diff
@@ -5,30 +5,22 @@
-      Pause
-    </button>
-    <button v-on:click="stop" :disabled="!isLoaded || isError || !isListening">
+    <button v-on:click="stop" :disabled="!isLoaded || error || !isListening">
```
I've been including "release" here too to kinda reset the demo, not sure if that's something we want to do though? I don't have any strong opinions

ksyeo1010 commented on README.md:
````diff
@@ -1421,70 +1421,56 @@ function VoiceWidget(props) {
 
 ```console
-yarn add @picovoice/porcupine-web-vue
+yarn add @picovoice/porcupine-vue @picovoice/web-voice-processor
````
yes, its a peer-dep so we should. the binding itself is broken if they dont install it.

ksyeo1010 commented on binding/vue/README.md:
````diff
@@ -1,127 +1,241 @@
 
-Install the package using `npm` or `yarn`. You will also need to add one of the `@picovoice/porcupine-web-**-worker` series of packages for the specific language model:
+```console
+yarn add @picovoice/porcupine-vue @picovoice/web-voice-processor
````
yes
